rs/ic-observability/multiservice-discovery/tests/tests.rs

```diff
@@ -0,0 +1,131 @@
+// Note: If you need to update the assertions, please check the README.md file for guidelines.
+#[cfg(test)]
+mod tests {
+    use assert_cmd::cargo::CommandCargoExt;
+    use multiservice_discovery_shared::builders::prometheus_config_structure::{
+        PrometheusStaticConfig, IC_NAME, IC_NODE, IC_SUBNET, JOB
+    };
+    use serde_json::Value;
+    use tempfile::tempdir;
+    use std::collections::{BTreeMap, BTreeSet};
+    use std::process::Command;
+    use std::time::Duration;
+    use std::thread;
+    use tokio::runtime::Runtime;
+
+    const BAZEL_SD_BIN: &str = "rs/ic-observability/multiservice-discovery/multiservice-discovery";
+    const CARGO_SD_BIN: &str = "multiservice-discovery";
+
+    async fn fetch_targets() -> anyhow::Result<Vec<PrometheusStaticConfig>> {
+        const TARGETS_URL: &str = "http://localhost:8000/prom/targets";
+        let timeout_duration = Duration::from_secs(500);
+        let start_time = std::time::Instant::now();
+
+        loop {
+            if start_time.elapsed() > timeout_duration {
+                return Err(anyhow::anyhow!("Timeout reached"));
+            }
+            if let Ok(response) = reqwest::get(TARGETS_URL).await {
+                if response.status().is_success() {
+                    let text = response.text().await?;
+                    let targets: Vec<PrometheusStaticConfig> = serde_json::from_str(&text).unwrap();
+                    return Ok(targets);
+                }
+            }
+            thread::sleep(Duration::from_secs(5));
+        }
+    }
+
+
+    async fn expected_nodes_and_subnets() -> Result<(Vec<String>, Vec<String>), reqwest::Error> {
+        const API_NODES_URL: &str = "https://ic-api.internetcomputer.org/api/v3/nodes";
+        let response = reqwest::get(API_NODES_URL).await?.text().await?;
```

Comment:
We should add a [timeout](https://docs.rs/reqwest/latest/reqwest/struct.RequestBuilder.html#method.timeout) and retries here, since the API may be partially or fully unavailable.